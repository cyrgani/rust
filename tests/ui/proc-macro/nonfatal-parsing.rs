//@ proc-macro: nonfatal-parsing.rs
//@ needs-unwind

extern crate nonfatal_parsing;
extern crate proc_macro;

#[path = "auxiliary/nonfatal-parsing-body.rs"]
mod body;

fn main() {
    nonfatal_parsing::run!();
    //~^ ERROR: invalid digit for a base 2 literal
    //~| ERROR: found invalid character; only `#` is allowed in raw string delimitation: \u{0}
    //~| ERROR: no valid digits found for number [E0768]
    //~| ERROR: binary float literal is not supported


    // FIXME: enable this once the standalone backend exists
    // body::run();
}
