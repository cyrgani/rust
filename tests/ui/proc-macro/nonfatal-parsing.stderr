error: invalid digit for a base 2 literal
  --> $DIR/nonfatal-parsing.rs:11:5
   |
LL |     nonfatal_parsing::run!();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `nonfatal_parsing::run` (in Nightly builds, run with -Z macro-backtrace for more info)

error: found invalid character; only `#` is allowed in raw string delimitation: \u{0}
  --> $DIR/nonfatal-parsing.rs:11:5
   |
LL |     nonfatal_parsing::run!();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `nonfatal_parsing::run` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0768]: no valid digits found for number
  --> $DIR/nonfatal-parsing.rs:11:5
   |
LL |     nonfatal_parsing::run!();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `nonfatal_parsing::run` (in Nightly builds, run with -Z macro-backtrace for more info)

error: binary float literal is not supported
  --> $DIR/nonfatal-parsing.rs:11:5
   |
LL |     nonfatal_parsing::run!();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `nonfatal_parsing::run` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0768`.
